activiti-cloud-service-common/activiti-cloud-services-dbp-rest/src/main/java/org/activiti/cloud/alfresco/argument/resolver/AlfrescoPageArgumentMethodResolver.java

```diff
@@ -71,4 +81,32 @@ public Pageable resolveArgument(
             return basePageable;
         }
     }
+
+    private boolean isPaginationValueExceedingLimit(
+        AlfrescoQueryParameters alfrescoQueryParameters,
+        Pageable basePageable
+    ) {
+        if (maxItemsLimitEnabled) {
+            if (
+                alfrescoQueryParameters.getMaxItemsParameter().isSet() &&
+                alfrescoQueryParameters.getMaxItemsParameter().getValue() > maxItemsLimit
+            ) {
+                return true;
+            }
+            if (
+                alfrescoQueryParameters.getPageSizeParameter().isSet() &&
+                alfrescoQueryParameters.getPageSizeParameter().getValue() > maxItemsLimit
+            ) {
+                return true;
+            } else if (
+                alfrescoQueryParameters.getMaxItemsParameter().isSet() &&
+                alfrescoQueryParameters.getMaxItemsParameter().getValue() < maxItemsLimit
+            ) {
+                return false;
+            } else {
+                return basePageable.getPageSize() >= maxItemsLimit;
+            }
+        }
+        return false;
```

Comment:
I think this entire block can be simplified to the code bellow, because there are only two cases that are problematic:
1. when the Alfresco naming convention is used (`maxItems` set): maxItems > maxItemsLimit
2. when Spring naming conversion is used (`maxItems` no set): pageSize > maxItemsLimit
```suggestion
        if (maxItemsLimitEnabled) {
            if (alfrescoQueryParameters.getMaxItemsParameter().isSet()) {
                return alfrescoQueryParameters.getMaxItemsParameter().getValue() > maxItemsLimit;
            } else {
                return basePageable.getPageSize() > maxItemsLimit;
            }
        }
        return false;
```